### PR TITLE
Add bittrex exchange support and unit test

### PR DESCRIPTION
**Pull request by naranjo074**
**Is this pull request adding a feature or fixing a bug?**  
It's add support for bittrex exchange, authentication, balances and unit test are mainly covers with pr.
[Bittrex issue](https://github.com/balancemymoney/balance-open/issues/66)

**Does this pull request close an issue? If so, which one?**
No

**What does this pull request do? What does it change?**
It's only add another exchange, mainly the interface to get access to bittrex API. This pr introduces a little change with the API, it's uses a new one with the intention of remove the old one using basic logic from this one.

@reddavis @einsteinx2, please feel free to comment, i would want feedback because i only add the interface to get access so may be I miss cover some features.

Unit test were done in another target, because the only one doesn't let target be compiled, it needs a provisioning profile with capabilities.

### CODE REVIEW

File: Balance/Shared/Data Model/Exchange Rates/ExchangeRateSource.swift [hunk 1]
```diff
@@ -17,6 +17,7 @@ public enum ExchangeRateSource: Int {
     case kucoin       = 5
     case hitbtc       = 6
     case binance      = 7
+    case bittrex      = 8
```
Comment (naranjo074) [inline Balance/Shared/Data Model/Exchange Rates/ExchangeRateSource.swift]: I am not sure about the changes done on this file, I would like guys if you give me more context. @einsteinx2 @wolffan

Comment (wolffan) [reply Balance/Shared/Data Model/Exchange Rates/ExchangeRateSource.swift]: that's ok, we just need to add bittrex as an exchange on the backend (and parse it's exchange data)

**Comment by einsteinx2** [comment]
@naranjo074 when you get a moment, can you click the "CLA Not Signed Yet" button? We need that before we can merge.

**Comment by jiashuw** [comment]
@naranjo074 hey any updates on this PR? what's next step?

**Comment by jiashuw** [comment]
💥

**Comment by naranjo074** [comment]
@jiashuw sorry for the delay, I only added an interface to get access to bittrex. So from now it's only to use the correct endpoints. This exchange needs to be showed to the user to let it use this. So I will be connecting the ui(Mac and iOS) as soon as possible. If you have any question let me know I will be glad giving an answer.

**Comment by einsteinx2** [comment]
Instead of reverting I'm just going to merge master back into develop

**Pull request merged**